plugins/content_distribution/lib/kContentDistributionFlowManager.php

```diff
@@ -1617,6 +1623,13 @@ public static function onEntryDeleted(entry $entry)
 				KalturaLog::log("Entry distribution [" . $entryDistribution->getId() . "] already flagged for deletion");
 				continue;
 			}
+			
+			if (!in_array ($entryDistribution->getStatus(), $validStatus))
+			{
+				$entryDistribution->setStatus(EntryDistributionStatus::DELETED);
+				$entryDistribution->setDirtyStatus(EntryDistributionDirtyStatus::NONE);
+				$entryDistribution->save ();
```

Comment:
First I think that UPDATING should behave the same and you should also `continue;` after this save, because as you said, the submitDelete will fail anyway.